test/unittests/test_excel_to_json_lists.py

```diff
@@ -25,20 +27,95 @@ def tearDownClass(cls) -> None:
             os.remove('testdata/tmp/' + file)
         os.rmdir('testdata/tmp')
 
-    def test_excel2jsonlist(self) -> None:
+
+    def test_expand_lists_from_excel(self) -> None:
+        # take the "lists" section of the systematic test project, expand it, and check if it is equal to the expanded
+        # version stored in the testdata folder
+        with open("testdata/test-project-systematic.json") as f:
+            lists_with_excel_reference = json.load(f)["project"]["lists"]
+        lists_with_excel_reference_output, success1 = e2l.expand_lists_from_excel(lists_with_excel_reference)
+        with open("testdata/lists_section_expanded.json") as f:
+            lists_with_excel_reference_output_expected = json.load(f)["expanded lists section of test-project-systematic.json"]
+        self.assertTrue(success1)
+        self.assertListEqual(lists_with_excel_reference_output, lists_with_excel_reference_output_expected)
+
+        # take the expanded version, and make sure that it is returned unchanged
+        lists_without_excel_reference = lists_with_excel_reference_output_expected
+        lists_without_excel_reference_output, success2 = e2l.expand_lists_from_excel(lists_without_excel_reference)
+        self.assertTrue(success2)
+        self.assertListEqual(lists_without_excel_reference, lists_without_excel_reference_output)
+
+
+    def test_make_json_lists_from_excel(self) -> None:
+        lists_multilingual = [f"testdata/lists_multilingual/{lang}.xlsx" for lang in ["de", "en", "fr"]]
+        lists_multilingual_output = e2l._make_json_lists_from_excel(lists_multilingual)
+        with open("testdata/lists_multilingual_output_expected.json") as f:
+            lists_multilingual_output_expected = json.load(f)["lists"]
+        self.assertListEqual(lists_multilingual_output, lists_multilingual_output_expected)
+
+
+    def test_validate_lists_section_with_schema(self) -> None:
+        with open("testdata/lists_multilingual_output_expected.json") as f:
+            lists_section_valid = json.load(f)["lists"]
+
+        # validate the valid "lists" section in a correct way
+        self.assertTrue(e2l.validate_lists_section_with_schema(lists_section=lists_section_valid))
+
+        # remove mandatory "comments" section from root node
+        lists_section_without_comment_at_rootnode = copy.deepcopy(lists_section_valid)
+        del lists_section_without_comment_at_rootnode[0]["comments"]
+        with self.assertRaisesRegex(
```

Comment:
Here (and in the other calls to `self.assertRaisesRegex()` I have a problem: As soon as the text of the error message slightly differs from the regex, I get a very strange behaviour: Instead of a proper message that the test failed, I get an infinitely long stack trace, and if I do it in debugging mode, Python unexpectedly quits due to threading problems.

This is not a problem for right now, because the tests run through, but it is not so nice that in case of failure, the error is difficult to track.

Reply:
I think we should use pytest instead of unittest anyways... and in pytest, testing for exceptions works a lot better than in unittest, I think. (I haven't tried it myself really, but I heard so.)